tests/15_Versioning/VersionManagerTest.php

```diff
@@ -275,6 +283,39 @@ public function testRestorePendingChanges()
         $this->vm->restore(true, $version);
     }
 
+    public function testRestoreIsCheckedIn()
+    {
+        $nodePath = '/tests_version_base/versioned';
+        $version = $this->vm->checkpoint($nodePath);
+        $this->vm->checkout($nodePath);
+        $this->assertTrue($this->vm->isCheckedOut($nodePath));
+
+        $this->vm->restore(true, $version);
+        $this->assertFalse($this->vm->isCheckedOut($nodePath));
+    }
+
+    public function testRestoreBaseProperties()
+    {
+        // TODO also check for primary node type and uuid once it can be changed
```

Comment:
I've got the same understanding until now, but after having a look at the specification I am not sure if there are some edge cases at which this properties can change. Otherwise 15.7.2 on http://www.day.com/specs/jcr/2.0/15_Versioning.html doesn't make any sense to me.